repos/fdbt-site/src/pages/api/updateFareTriangle.ts

```diff
@@ -0,0 +1,158 @@
+import { NextApiResponse } from 'next';
+import { putDataInS3 } from '../../data/s3';
+import { NextApiRequestWithSession, ErrorInfo, UserFareStages } from '../../interfaces';
+import { redirectToError, redirectTo } from '../../utils/apiUtils';
+import {
+    INPUT_METHOD_ATTRIBUTE,
+    CSV_UPLOAD_ATTRIBUTE,
+    MATCHING_JSON_ATTRIBUTE,
+    PRODUCT_AND_SERVICE_ID_ATTRIBUTE,
+} from '../../constants/attributes';
+import { getFormData, processFileUpload } from '../../utils/apiUtils/fileUpload';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import { WithIds, ReturnTicket, SingleTicket } from '../../../shared/matchingJsonTypes';
+import { faresTriangleDataMapper, setCsvUploadAttributeAndRedirect } from './csvUpload';
+
+const errorId = 'csv-upload';
+
+export const config = {
+    api: {
+        bodyParser: false,
+    },
+};
+
+const getNamesOfFareZones = (ticket: WithIds<SingleTicket> | WithIds<ReturnTicket>) => {
+    let fareZoneNames;
+
+    if ('fareZones' in ticket) {
+        fareZoneNames = ticket.fareZones.map((fz) => fz.name);
+    } else if ('inboundFareZones' in ticket && 'outboundFareZones' in ticket) {
+        const combinedFareZones = ticket.outboundFareZones.concat(ticket.inboundFareZones);
+
+        const combinedNamesOnly = combinedFareZones.map((x) => x.name);
+
+        // get rid of duplicates
+        fareZoneNames = [...new Set(combinedNamesOnly)];
+    }
+
+    return fareZoneNames;
+};
+
+const checkFareStageNames = (fareTriangleData: UserFareStages, fareZoneNames: string[]): boolean => {
```

Comment:
Could we make a quick unit test for this? I think only one scenario needed.

Reply:
Will add unit test.

Reply:
Unit test done :+1: